gravity-forms/gw-set-list-field-rows-by-field-value.php

```diff
@@ -16,146 +16,147 @@
- 
+
 	}
- 
+
 	public function load_form_script( $form, $is_ajax_enabled ) {
- 
-		if( $this->is_applicable_form( $form ) && ! has_action( 'wp_footer', array( $this, 'output_script' ) ) ) {
+
+		if ( $this->is_applicable_form( $form ) && ! has_action( 'wp_footer', array( $this, 'output_script' ) ) ) {
 			add_action( 'wp_footer', array( $this, 'output_script' ) );
 			add_action( 'gform_preview_footer', array( $this, 'output_script' ) );
 		}
- 
+
 		return $form;
 	}
- 
+
 	public function output_script() {
 		?>
- 
-		<script type="text/javascript">
 
-            window.gwalfr;
+		<script type="text/javascript">
 
-            (function($){
+			window.gwalfr;
 
-                gwalfr = function( args ) {
+			(function($){
+				var gf25 = <?php echo ( version_compare( GFForms::$version, '2.5.0', '>=' ) ) ? 'true' : 'false'; ?>;
+				gwalfr = function( args ) {
 
-                    this.formId      = args.formId,
-                        this.listFieldId = args.listFieldId,
-                        this.inputHtmlId = args.inputHtmlId;
+					this.formId      = args.formId,
+						this.listFieldId = args.listFieldId,
+						this.inputHtmlId = args.inputHtmlId;
 
-                    this.init = function() {
+					this.init = function() {
 
-                        var gwalfr = this,
-                            triggerInput = $( this.inputHtmlId );
+						var gwalfr = this,
+							triggerInput = $( this.inputHtmlId );
 
-                        // update rows on page load
-                        this.updateListItems( triggerInput, this.listFieldId, this.formId );
+						// update rows on page load
+						this.updateListItems( triggerInput, this.listFieldId, this.formId );
 
-                        // update rows when field value changes
-                        triggerInput.change(function(){
-                            gwalfr.updateListItems( $(this), gwalfr.listFieldId, gwalfr.formId );
-                        });
+						// update rows when field value changes
+						triggerInput.change(function(){
+							gwalfr.updateListItems( $(this), gwalfr.listFieldId, gwalfr.formId );
+						});
 
-                    }
+					}
 
-                    this.updateListItems = function( elem, listFieldId, formId ) {
+					this.updateListItems = function( elem, listFieldId, formId ) {
 
-                        var listField = $( '#field_' + formId + '_' + listFieldId ),
-                            count = parseInt( elem.val() );
-                        rowCount = listField.find( 'table.gfield_list tbody tr' ).length,
-                            diff = count - rowCount;
+						var listField = $( '#field_' + formId + '_' + listFieldId ),
+							count = parseInt( elem.val() );
+						// GF 2.5 doesn't use tables by default, look for `gfield_list_group` in that case with GF2.4 fallback
+						rowCount = ( gf25 ) ? listField.find('.gfield_list_group').length : listField.find( 'table.gfield_list tbody tr' ).length,
```

Comment:
I believe this can be simplified to just `listField.find('.gfield_list_group').length` which should work with both GF 2.4 and 2.5.

Reply:
@claygriffiths Great call! I didn't check GF 2.4 to confirm if that would work there too.

I went ahead and removed the 2.5 check and simplified the rowCount setup.